### ISSUE TITLE
the manual step for shared is failing with error

### ISSUE THREAD

**Issue by kumarvt** [terraform-google-modules/terraform-example-foundation#643]
### TL;DR

I'm running the manual step as 
https://github.com/terraform-google-modules/terraform-example-foundation/tree/develop/3-networks/README.md

1. You will need only once to manually plan + apply the `shared` environment since dev, nonprod and prod depend on it.
    1. cd to ./envs/shared/
    1. Update backend.tf with your bucket name from the bootstrap step.
    1. Run `terraform plan` and review output

### Expected behavior

if the project_id is not advised to be set in documentation in any of the tfvars files than this process should complete successfully.

### Observed behavior

Getting these errors
    `~/.../envs/shared$ terraform plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

data.google_service_account_access_token.default: Refreshing state...
data.google_active_folder.non-production: Refreshing state...
data.google_active_folder.development: Refreshing state...
data.google_active_folder.production: Refreshing state...
data.google_active_folder.common: Refreshing state...
data.google_active_folder.bootstrap: Refreshing state...
data.google_projects.dns_hub: Refreshing state...
data.google_projects.restricted_net_hub[0]: Refreshing state...
data.google_projects.base_net_hub[0]: Refreshing state...

------------------------------------------------------------------------

Error: Error in function call

  on ../../modules/restricted_shared_vpc/service_control.tf line 51, in resource "google_access_context_manager_service_perimeter" "regular_service_perimeter":
  51:     resources           = formatlist("projects/%s", [var.project_number])
    |----------------
    | var.project_number is null

Call to function "formatlist" failed: error on format iteration 0: unsupported
value for "%s" at 9: null value cannot be formatted.

Error: Missing required argument

  on .terraform/modules/base_shared_vpc.main/modules/vpc/main.tf line 37, in resource "google_compute_shared_vpc_host_project" "shared_vpc_host":
  37:   project    = var.project_id

The argument "project" is required, but no definition was found.

Error: Missing required argument

  on .terraform/modules/restricted_shared_vpc.main/modules/vpc/main.tf line 37, in resource "google_compute_shared_vpc_host_project" "shared_vpc_host":
  37:   project    = var.project_id

The argument "project" is required, but no definition was found.
`

### Terraform Configuration

```hcl
i've been able to successfully run through all previous steps from bootstrap to environments
```

### Terraform Version

```sh
$:~/.../envs/shared$ terraform --version
Terraform v0.13.7
+ provider registry.terraform.io/hashicorp/google v3.90.1
+ provider registry.terraform.io/hashicorp/google-beta v3.90.1
+ provider registry.terraform.io/hashicorp/null v3.1.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
```

**Comment by kumarvt** [comment]
After a bit of cheking and trying , i have been able to run the tf plan step as above, i'm yet to understand whats wrong with having the hub_and_spoke option enabled

by commenting in ./gcp-networks/envs/development/common.auto.tfvars
// enable_hub_and_spoke = true
